DigitalLearningSolutions.Data.Tests/Services/GroupServiceTests/GroupsServiceTests.cs

```diff
@@ -385,6 +447,279 @@ public void
             A.CallTo(() => groupsDataService.GetGroupCoursesForCentre(2)).MustHaveHappenedOnceExactly();
         }
 
+        [Test]
+        public void
+            GenerateGroupsFromRegistrationField_calls_data_service_methods_with_correct_values_for_custom_prompt()

```

Comment:
There are several methods here for the new GenerateGroups service method that have similar/duplicate setups (e.g. time, GetGroupsForCentre, AddingTheDelegates etc.) and assertions. Can we commonise all of these to reduce the length of these tests.